pdns/pdns_recursor.cc

```diff
@@ -1320,15 +1317,15 @@ static void startDoResolve(void *p)
     auto luaconfsLocal = g_luaconfs.getLocal();
     // Used to tell syncres later on if we should apply NSDNAME and NSIP RPZ triggers for this query
     bool wantsRPZ(true);
-    boost::optional<RecProtoBufMessage> pbMessage(boost::none);
+    std::unique_ptr<pdns::ProtoZero::Message> pbMessage;
```

Comment:
Perhaps we could make that a `boost::optional<pdns::ProtoZero::Message>` instead, to prevent a dynamic memory allocation?

Reply:
My original code had that but it caused bugs. Will investigate a bit further now general part is working.

Reply:
I think it had to do with the default constructor of `protozero::pbf_writer` being used, which creates a non-working writer (as documented). So I'd have to see if that can be avoided somehow even if  the object is embedded in a `boost::optional`